examples/common_transformations.rs

```diff
@@ -0,0 +1,151 @@
+use signal_hook::{consts, iterator::Signals};
+use std::{
+    process::{Command, Stdio},
+    thread,
+    time::Duration,
+};
+use video_compositor::http;
+
+use crate::common::write_example_sdp_file;
+
+#[path = "./common/common.rs"]
+mod common;
+
+const VIDEO_RESOLUTION: Resolution = Resolution {
+    width: 1920,
+    height: 1080,
+};
+
+const FRAMERATE: Framerate = Framerate(30);
+
+fn main() {
+    env_logger::init_from_env(
+        env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+    );
+    ffmpeg_next::format::network::init();
+
+    thread::spawn(|| {
+        if let Err(err) = start_example_client_code() {
+            error!("{err}")
+        }
+    });
+
+    http::Server::new(8001).start();
+
+    let mut signals = Signals::new([consts::SIGINT]).unwrap();
+    signals.forever().next();
+}
+
+fn start_example_client_code() -> Result<()> {
+    thread::sleep(Duration::from_secs(2));
+
+    info!("[example] Sending init request.");
+    common::post(&json!({
+        "type": "init",
+        "framerate": FRAMERATE,
+        "init_web_renderer": false,
+    }))?;
+
+    info!("[example] Start listening on output port.");
+    let output_sdp = write_example_sdp_file("127.0.0.1", 8002)?;
+    Command::new("ffplay")
+        .args(["-protocol_whitelist", "file,rtp,udp", &output_sdp])
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()?;
+
+    info!("[example] Send register output request.");
+    common::post(&json!({
+        "type": "register_output",
+        "id": "output 1",
+        "port": 8002,
+        "ip": "127.0.0.1",
+        "resolution": {
+            "width": VIDEO_RESOLUTION.width,
+            "height": VIDEO_RESOLUTION.height,
+        },
+        "encoder_settings": {
+            "preset": "ultrafast"
+        }
+    }))?;
+
+    info!("[example] Send register input request.");
+    common::post(&json!({
+        "type": "register_input",
+        "id": "input 1",
+        "port": 8004
+    }))?;
+
+    info!("[example] Register static image");
+    common::post(&json!({
+        "type": "register_transformation",
+        "key": "example_image",
+        "transform": {
+            "type": "image",
+            "asset_type": "jpeg",
+            "url": "https://i.postimg.cc/NfkxF1SV/wp5220836.jpg",
+        }
+    }))?;
+
+    info!("[example] Update scene");
+    common::post(&json!({
+        "type": "update_scene",
+        "inputs": [],
+        "transforms": [
+            {
+                "node_id": "image",
+                "type": "image",
+                "image_id": "example_image",
+           },
+            {
+                "node_id": "fitted",
+                "type": "common",
+                "transformation": {
+                    "type": "convert_resolution",
+                    "params": {
+                        "strategy": "fill_scale",
+                        "color": "#FFFFFF00"
+                    },
+                },
```

Comment:
```suggestion
                "type": "built-in:convert_resolution",
                "params": {
                    "strategy": "fill_scale",
                    "color": "#FFFFFF00"
                },
```

try to avoid nesting if it's not necessarry, I would even consider
```suggestion
                "type": "built-in:convert_resolution",
                "strategy": "fill_scale",
                "color": "#FFFFFF00"
```

Reply:
Since implementing this would require nontrivial parsing, I implemented it as:
```json
"node_id": "fitted",
"type": "built-in",
"transformation": "transform_to_resolution",
"strategy": "fit",
"color": "#FFFFFF00",
"resolution": { "width": VIDEO_RESOLUTION.width, "height": VIDEO_RESOLUTION.height },
"input_pads": ["image"],
```